gnoland/website/pkgs/doc/doc_test.go

```diff
@@ -0,0 +1,178 @@
+	}
+
+	if len(pkg.Filenames) != 1 {
+		t.Errorf("package filenames: got %d, want 1 file", len(pkg.Filenames))
+	}
+
+	if len(pkg.Consts) != 1 {
+		t.Errorf("package consts: got %d, want 1 const", len(pkg.Consts))
+	}
+
+	if len(pkg.Vars) != 2 {
+		t.Errorf("package vars: got %d, want 2 vars", len(pkg.Vars))
+	}
+
+	if len(pkg.Funcs) != 5 {
+		t.Errorf("package funcs: got %d, want 5 functions", len(pkg.Funcs))
+	}
+
+	if len(pkg.Types) != 2 {
+		t.Errorf("package types: got %d, want 2 types", len(pkg.Types))
+	} else {
+		myInterfaceType := pkg.Types[0]
+		if want := "MyInterface"; myInterfaceType.Name != want {
+			t.Errorf("package type name: got %q, want %q", myInterfaceType.Name, want)
+		}
+		if want := "MyInterface"; myInterfaceType.ID != want {
+			t.Errorf("package type id: got %q, want %q", myInterfaceType.Name, want)
+		}
+		if want := "A public interface.\n"; myInterfaceType.Doc != want {
+			t.Errorf("package type doc: got %q, want %q", myInterfaceType.Doc, want)
+		}
+		if want := 0; len(myInterfaceType.Funcs) != want {
+			t.Errorf("package type funcs: got %d, want %d", len(myInterfaceType.Funcs), want)
+		}
+		if want := 0; len(myInterfaceType.Methods) != want {
+			t.Errorf("package type methods: got %d, want %d", len(myInterfaceType.Methods), want)
+		}
+		if want := 0; len(myInterfaceType.Vars) != want {
+			t.Errorf("package type vars: got %d, want %d", len(myInterfaceType.Vars), want)
+		}
+		if want := 0; len(myInterfaceType.Consts) != want {
+			t.Errorf("package type vars: got %d, want %d", len(myInterfaceType.Consts), want)
+		}
+		if want := "type MyInterface interface {\n\tMyMethod() string\n}"; myInterfaceType.Definition != want {
+			t.Errorf("package type definition: got %q, want %q", myInterfaceType.Definition, want)
+		}
+
+		myTypeType := pkg.Types[1]
+		if want := "MyType"; myTypeType.Name != want {
+			t.Errorf("package type name: got %q, want %q", myTypeType.Name, want)
+		}
+		if want := "MyType"; myTypeType.ID != want {
+			t.Errorf("package type id: got %q, want %q", myTypeType.Name, want)
+		}
+		if want := "A public type.\n"; myTypeType.Doc != want {
+			t.Errorf("package type doc: got %q, want %q", myTypeType.Doc, want)
+		}
+		if want := 2; len(myTypeType.Methods) != want {
+			t.Errorf("package type methods: got %d, want %d", len(myTypeType.Methods), want)
+		}
+		if want := 1; len(myTypeType.Funcs) != want {
+			t.Errorf("package type funcs: got %d, want %d", len(myTypeType.Funcs), want)
+		}
+		if want := 0; len(myTypeType.Vars) != want {
+			t.Errorf("package type vars: got %d, want %d", len(myTypeType.Vars), want)
+		}
+		if want := 0; len(myTypeType.Consts) != want {
+			t.Errorf("package type consts: got %d, want %d", len(myTypeType.Consts), want)
+		}
+		if want := "type MyType struct {\n\tName string // Name is a public field\n\t// contains filtered or unexported fields\n}"; myTypeType.Definition != want {
+			t.Errorf("package type definition: got %q, want %q", myTypeType.Definition, want)
+		}
```

Comment:
i don't want to be overly pedantic with tests, but this seems like a lot of code.

since we use `github.com/stretchr/testify/assert` (and require) in other packages, can you refactor this to use that?